sources/prairiedog/src/main.rs

```diff
@@ -0,0 +1,359 @@
+/*!
+  prairiedog is a tool to provide kdump support in Bottlerocket. It performs three operations:
+
+  - _digs_ to find the active boot partition and mounts it in /boot
+  - loads the crash kernel from /boot
+  - creates memory dumps when the kernel panics
+*/
+
+#![deny(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use argh::FromArgs;
+use nix;
+use signpost::State;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger, WriteLogger};
+use snafu::{ensure, ResultExt};
+use std::ffi::OsStr;
+use std::fs::{self, File};
+use std::path::Path;
+use std::process::{self, Command};
+
+// Kdump related binary paths
+const MAKEDUMPFILE_PATH: &str = "/sbin/makedumpfile";
+const KEXEC_PATH: &str = "/sbin/kexec";
+
+// Mount points created prairiedog
+const BOOT_MOUNT_PATH: &str = "/boot";
+
+// Mounts configurations
+const BOOT_MOUNT_SELINUX_CONTEXT: &str = "context=system_u:object_r:os_t:s0";
+
+// Files generated by prairiedog
+const KDUMP_LOGS_PATH: &str = "/var/log/kdump";
+const LOG_FILE: &str = "prairiedog.log";
+const DMESG_DUMP_FILE: &str = "dmesg.dump";
```

Comment:
Since it's plain text:
```suggestion
const DMESG_DUMP_FILE: &str = "dmesg.log";
```